core/src/consensus/tendermint/network.rs

```diff
@@ -157,24 +159,22 @@ impl TendermintExtension {
         }
     }
 
-    fn request_proposal_to_any(&self, round: SortitionRound) {
-        for (token, peer) in &self.peers {
-            let is_future_height_and_view = {
-                let higher_height = peer.vote_step.height > round.height;
-                let same_height_and_higher_view =
-                    peer.vote_step.height == round.height && peer.vote_step.view > round.view;
-                higher_height || same_height_and_higher_view
-            };
-
-            if is_future_height_and_view {
-                self.request_proposal(token, round);
-                continue
-            }
-
-            let is_same_height_and_view = peer.vote_step.height == round.height && peer.vote_step.view == round.view;
-
-            if is_same_height_and_view && peer.proposal.is_some() {
-                self.request_proposal(token, round);
+    fn request_proposal_to_superiors(&self, round: SortitionRound, my_highest: Option<Priority>) {
+        if let Some(peer_highest_priority) = self.peers.iter().map(|(_id, peer)| peer.priority).max() {
+            let highest_priority_peers = self.peers.iter().filter(|(_id, peer)| peer.priority == peer_highest_priority);
```

Comment:
Ok, that makes sense! I'll think about it

Reply:
Hmm.. I need to come up with an efficient way to avoid double verification. One is against `peer_state_message` and the other is against `on_proposal_message`